core/src/test/java/com/google/cloud/sql/core/DefaultAccessTokenSupplierTest.java

```diff
@@ -427,4 +440,48 @@ public void testGetTokenExpiration() {
                 Optional.of(new AccessToken("", past))))
         .isEqualTo(Optional.of(past));
   }
+
+  private static class Oauth2CredentialFactory implements CredentialFactory {
+    private final Credential credential;
+
+    private Oauth2CredentialFactory(Credential credential) {
+      this.credential = credential;
+    }
+
+    @Override
+    public HttpRequestInitializer create() {
+      return credential;
+    }
+  }
+
+  private static class Oauth2BadCredentialFactory implements CredentialFactory {
+    private final OAuth2Credentials credential;
+
+    private Oauth2BadCredentialFactory(OAuth2Credentials credential) {
+      this.credential = credential;
+    }
+
+    @Override
+    public HttpRequestInitializer create() {
```

Comment:
This simulates an old implementation of CredentialFactory from before `getCredentials()` was introduced that returns the wrong kind of credential in an HttpRequestInitializer.